airflow/jobs.py

```diff
@@ -535,7 +535,10 @@ def process_dag(self, dag, queue):
             elif ti.is_runnable(flag_upstream_failed=True):
```

Comment:
I had not taken a deep look at the deadlock handling and `could_not_run` before. Now I understand how continue differs from `could_not_run.add`.